daemon/src/org/eclipse/iofog/element/Element.java

```diff
@@ -132,8 +130,8 @@ public void setVolumeMappings(List<VolumeMapping> volumeMappings) {
 
 	@Override
 	public boolean equals(Object e) {
```

Comment:
perhaps it's preferable to write like this:
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Element element = (Element) o;
        return Objects.equal(elementId, element.elementId);

Reply:
I will check it

Reply:
Josh Bloch favors your approach:

The reason that I favor the instanceof approach is that when you use the getClass approach, you have the restriction that objects are only equal to other objects of the same class, the same run time type. If you extend a class and add a couple of innocuous methods to it, then check to see whether some object of the subclass is equal to an object of the super class, even if the objects are equal in all important aspects, you will get the surprising answer that they aren't equal. In fact, this violates a strict interpretation of the Liskov substitution principle, and can lead to very surprising behavior. In Java, it's particularly important because most of the collections (HashTable, etc.) are based on the equals method. If you put a member of the super class in a hash table as the key and then look it up using a subclass instance, you won't find it, because they are not equal.